frontend-react/src/components/Admin/EditReceiverSettings.tsx

```diff
@@ -27,172 +25,153 @@ import {
     TextAreaComponent,
     TextInputComponent,
 } from "./AdminFormEdit";
+import { AdminFormWrapper } from "./AdminFormWrapper";
 
-type Props = {
+type EditReceiverSettingsFormProps = {
     orgname: string;
     receivername: string;
-    action: "edit" | "clone";
+    action: string;
 };
 
-export function EditReceiverSettings({ match }: RouteComponentProps<Props>) {
-    const orgname = match?.params?.orgname || "";
-    const receivername = match?.params?.receivername || "";
-    const action = match?.params?.action || "";
+const EditReceiverSettingsForm: React.FC<EditReceiverSettingsFormProps> = ({
+    orgname,
+    receivername,
+    action,
+}) => {
+    const [loading, setLoading] = useState(false);
+    const history = useHistory();
+    const confirmModalRef = useRef<ConfirmSaveSettingModalRef>(null);
 
-    const FormComponent = () => {
-        const [loading, setLoading] = useState(false);
-        const history = useHistory();
-        const confirmModalRef = useRef<ConfirmSaveSettingModalRef>(null);
+    const orgReceiverSettings: OrgReceiverSettingsResource = useResource(
+        OrgReceiverSettingsResource.detail(),
+        { orgname, receivername, action }
+    );
 
-        const orgReceiverSettings: OrgReceiverSettingsResource = useResource(
-            OrgReceiverSettingsResource.detail(),
-            { orgname, receivername, action }
-        );
+    const { fetch: fetchController } = useController();
+    const [orgReceiverSettingsOldJson, setOrgReceiverSettingsOldJson] =
+        useState("");
+    const [orgReceiverSettingsNewJson, setOrgReceiverSettingsNewJson] =
+        useState("");
+    const { invalidate } = useController();
 
-        const { fetch: fetchController } = useController();
-        const [orgReceiverSettingsOldJson, setOrgReceiverSettingsOldJson] =
-            useState("");
-        const [orgReceiverSettingsNewJson, setOrgReceiverSettingsNewJson] =
-            useState("");
-        const { invalidate } = useController();
+    async function getLatestReceiverResponse() {
+        const accessToken = getStoredOktaToken();
```

Comment:
Out of scope on this ticket, but @TomNUSDS I believe this is where our Okta token refresh woes stem from. Not this single instance, just this pattern of using `getStoredOktaToken`. `@okta/react` has an OktaAuth hook we can use that, I think, will handle that refresh when necessary.

Reply:
Excellent point. 

But is refreshing the token an async operation to the otka server? How does a request flow go if it has to first wait for a reply from Okta for a refresh?

Maybe there should be a global timer running in the app that tries to refresh the stored token every 10 minutes IF there's been activity. (Defining activity is another problem).